### ISSUE TITLE
Enhancement: [no-type-alias] Allow more cases

### ISSUE THREAD

**Issue by somebody1234** [typescript-eslint/typescript-eslint#6588]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-type-alias/

### Description

I propose that the `no-type-alias` rule should handle more cases:
- `typeof objectValue`
- namespace re-exports: `export type Foo<T> = MyNamespace.Bar<T>`

there may be more cases that should also be handled separately, but these are the ones I've run into so far

### Fail

```typescript
N/A
```

### Pass

```typescript
export type Foo = typeof foo
export type Bar<T> = MyNamespace.Bar<T>
```

**Comment by JoshuaKGoldberg** [comment]
The `no-type-alias` rule is pretty old and we don't recommend users use it anymore. I fact, we're about to deprecate it for v6:

* #6036
* #6229

Thanks for filing - but we're not planning on adding to this rule.
